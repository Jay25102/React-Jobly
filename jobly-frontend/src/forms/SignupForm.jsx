import React, { useState } from "react";
import { redirect } from "react-router-dom";
import Alert from "../Alert";

/**
 * Form to register a new user.
 * 
 * When submitted, calls signup function prop defined in App
 * then redirects to /companies
 */
function SignupForm({signup}) {
    const initialState = {
        username: "",
        password: "",
        firstName: "",
        lastName: "",
        email: "",
    };
    const [formData, setFormData] = useState(initialState);
    const [formErrors, setFormErrors] = useState([]);

    console.debug(
        "SignupForm",
        "signup=", typeof signup,
        "formData=", formData,
        "formErrors=", formErrors,
    )

    async function handleSubmit(e) {
        e.preventDefault();
        let result = await signup(formData);
        if (result.success) {
            return redirect("/companies");
        }
        else {
            setFormErrors(result.errors);
        }
    }

    function handleChange(e) {
        const { name, value } = e.target;
        setFormData(data => ({ ... data, [name]: value}));
    }

    return (
        <div className="SignupForm container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
            <h3 className="mb-3">Sign Up</h3>
            <div className="card">
                <div className="card-body">
                    <form onSubmit={handleSubmit}>
                        <div className="form-group">
                            <label>Username</label>
                            <input 
                                name="username"
                                className="form-control"
                                value={formData.username}
                                onChange={handleChange}
                                autoComplete="username"
                                required
                            />
                        </div>
                        <div className="form-group">
                            <label>Password</label>
                            <input 
                                type="password"
                                name="password"
                                className="form-control"
                                value={formData.password}
                                onChange={handleChange}
                                autoComplete="current-password"
                                required
                            />
                        </div>
                        <div className="form-group">
                            <label>First Name</label>
                            <input 
                                name="firstName"
                                className="form-control"
                                value={formData.firstName}
                                onChange={handleChange}
                                required
                            />
                        </div>
                        <div className="form-group">
                            <label>Last Name</label>
                            <input 
                                name="lastName"
                                className="form-control"
                                value={formData.lastName}
                                onChange={handleChange}
                                required
                            />
                        </div>
                        <div className="form-group">
                            <label>Email</label>
                            <input 
                                name="email"
                                className="form-control"
                                value={formData.email}
                                onChange={handleChange}
                                required
                            />
                        </div>

                        {formErrors.length
                                ? <Alert type="danger" messages={formErrors}/>
                                : null}

                        <button 
                            className="btn btn-primary float-right" 
                            type="submit" 
                            onSubmit={handleSubmit}
                        >Submit</button>
                </form>
                </div>
            </div>
            
        </div>
    )
}

export default SignupForm;