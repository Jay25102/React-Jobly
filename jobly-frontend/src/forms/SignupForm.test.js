import React from "react";
import { render } from "@testing-library/react";
import SignupForm from "./SignupForm";

describe("SignupForm", function() {
    test("it renders without crashing", function() {
        render(<SignupForm/>);
    });
    test("it matches snapshot", function() {
        const { asFragment } = render(<SignupForm/>);
        expect(asFragment()).toMatchSnapshot();
    });
});