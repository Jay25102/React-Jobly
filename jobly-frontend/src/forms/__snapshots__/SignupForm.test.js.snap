// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Sign Up
    </div>
    <form>
      <div>
        <label>
          Username
        </label>
        <input
          autocomplete="username"
          name="username"
          required=""
          value=""
        />
      </div>
      <div>
        <label>
          Password
        </label>
        <input
          autocomplete="current-password"
          name="password"
          required=""
          type="password"
          value=""
        />
      </div>
      <div>
        <label>
          First Name
        </label>
        <input
          name="firstName"
          required=""
          value=""
        />
      </div>
      <div>
        <label>
          Last Name
        </label>
        <input
          name="lastName"
          required=""
          value=""
        />
      </div>
      <div>
        <label>
          Email
        </label>
        <input
          name="email"
          required=""
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
